modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/SseEntryPoint.java

```diff
@@ -0,0 +1,169 @@
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.enonic.xp.app.Application;
+import com.enonic.xp.app.ApplicationKey;
+import com.enonic.xp.app.ApplicationNotFoundException;
+import com.enonic.xp.app.ApplicationService;
+import com.enonic.xp.event.Event;
+import com.enonic.xp.event.EventListener;
+import com.enonic.xp.impl.server.rest.model.ApplicationInfoJson;
+import com.enonic.xp.impl.server.rest.model.ApplicationUninstalledJson;
+import com.enonic.xp.impl.server.rest.model.ListApplicationJson;
+import com.enonic.xp.jaxrs.JaxRsComponent;
+
+@Path("sse")
+@Component(immediate = true, property = "group=api")
+public class SseEntryPoint
+    implements JaxRsComponent, EventListener
+{
+
+    private static final Logger LOG = LoggerFactory.getLogger( SseEntryPoint.class );
+
+    private static final String EVENT_TYPE = "application.cluster";
+
+    private static final String EVENT_TYPE_KEY = "eventType";
+
+    private static final String INSTALLED = "installed";
+
+    private static final String START = "start";
+
+    private static final String STOP = "stop";
+
+    private static final String UNINSTALL = "uninstall";
+
+    private static final String UNINSTALLED = "uninstalled";
+
+    private static final String APPLICATION_KEY_PARAM = "key";
+
+    private final ApplicationService applicationService;
+
+    private SseBroadcaster sseBroadcaster;
+
+    private Sse sse;
+
+    @Activate
+    public SseEntryPoint( final @Reference ApplicationService applicationService )
+    {
+        this.applicationService = applicationService;
+    }
+
+    @Deactivate
+    public void deactivate()
+    {
+        if ( sseBroadcaster != null )
+        {
+            sseBroadcaster.close();
+        }
+    }
+
+    @Context
+    public void setSse( final Sse sse )
+    {
+        this.sse = sse;
+        this.sseBroadcaster = sse.newBroadcaster();
+    }
+
+    @GET
+    @Path("subscribe-app-events")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public void subscribe( @Context SseEventSink sseEventSink )
+    {
+        this.sseBroadcaster.register( sseEventSink );
+
+        final OutboundSseEvent sseEvent = sse.newEventBuilder().
+            name( "List of applications" ).
+            id( UUID.randomUUID().toString() ).
+            mediaType( MediaType.APPLICATION_JSON_TYPE ).
+            data( ListApplicationJson.class, new ListApplicationJson( applicationService.getInstalledApplications().stream().
+                map( application -> new ApplicationInfoJson( application, applicationService.isLocalApplication( application.getKey() ) ) ).
+                collect( Collectors.toList() ) ) ).
+            comment( "List of installed applications" ).build();
+
+        sseEventSink.send( sseEvent );
+    }
+
+    @Override
+    public void onEvent( final Event event )
+    {
+        if ( EVENT_TYPE.equals( event.getType() ) )
```

Comment:
We maybe doing something wrong here. These events are global, but we probably need local (from bundle events)
Need to discuss.

Reply:
Done